corehq/motech/repeaters/expression/repeater_generators.py

```diff
@@ -14,3 +18,24 @@ def content_type(self):
     def get_payload(self, repeat_record, payload_doc, parsed_expression):
         result = parsed_expression(payload_doc, EvaluationContext(payload_doc))
         return json.dumps(result, cls=DjangoJSONEncoder)
+
+    def get_url(self, repeat_record, url_template, payload_doc):
+        if not toggles.UCR_EXPRESSION_REGISTRY.enabled(repeat_record.domain):
+            return ""
+
+        required_template_vars = [fn for _, fn, _, _ in Formatter().parse(url_template) if fn is not None]
+        context = EvaluationContext(payload_doc)
+        expressions = {
+            expression.name: expression.wrapped_definition(context)
+            for expression in UCRExpression.objects.filter(
+                domain=repeat_record.domain,
+                name__in=required_template_vars,
+                expression_type=UCR_NAMED_EXPRESSION,
+            )
+        }
+        return url_template.format(
+            **{
+                template_var: expressions[template_var](payload_doc) if template_var in expressions else ""
+                for template_var in required_template_vars
+            }
+        )
```

Comment:
Regarding @snopoke 's concerns about invalid URLs: What if failed loudly if `template_var` is not in `expressions`? Instead of allowing undefined template variables, we could require users to use a constant expression that returns an empty string? Or do you reckon that's being too strict / an unnecessary burden?